sql: validate RLS policy exemptions for constraints

spilchen opened the pull request:
Unique and foreign key constraints on tables with row-level security (RLS) enabled are exempt from RLS policies when enforcing constraints. This exemption was implemented previously. This commit adds tests to validate that RLS policies do not interfere with constraint enforcement.

Closes #136747

Epic: CRDB-45203
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143079)
<!-- Reviewable:end -->

rafiss reviewed (approved):
lgtm! as a possible additional test, would it be worth testing a CHECK constraint as well?

spilchen committed:
sql: validate RLS policy exemptions for constraints

Unique and foreign key constraints on tables with row-level security
(RLS) enabled are exempt from RLS policies when enforcing constraints.
This exemption was implemented previously. This commit adds tests to
validate that RLS policies do not interfere with constraint enforcement.

Closes #136747

Epic: CRDB-45203
Release note: none

spilchen commented:
> lgtm! as a possible additional test, would it be worth testing a CHECK constraint as well?

Good idea. I added a subtest to mix CHECK constraints and RLS policies.

spilchen commented:
TFTR!

bors r+
